packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -365,6 +374,650 @@ describe("Hardhat module", function () {
           });
         }
       });
+
+      describe("hardhat_setBalance", function () {
+        it("should reject an invalid address", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setBalance",
+            ["0x1234", "0x0"],
+            'Errors encountered in param 0: Invalid value "0x1234" supplied to : ADDRESS'
+          );
+        });
+
+        it("should reject a non-numeric balance", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setBalance",
+            [DEFAULT_ACCOUNTS_ADDRESSES[0].toString(), "xyz"],
+            'Errors encountered in param 1: Invalid value "xyz" supplied to : QUANTITY'
+          );
+        });
+
+        it("should not reject valid argument types", async function () {
+          await this.provider.send("hardhat_setBalance", [
+            DEFAULT_ACCOUNTS_ADDRESSES[0].toString(),
+            "0x0",
+          ]);
+        });
+
+        it("should result in a modified balance", async function () {
+          // Arrange: Capture existing balance
+          const existingBalance = rpcQuantityToBN(
+            await this.provider.send("eth_getBalance", [
+              DEFAULT_ACCOUNTS_ADDRESSES[0],
+            ])
+          );
+
+          // Act: Set the new balance.
+          const targetBalance = existingBalance.add(new BN(1)).mul(new BN(2));
+          // For sanity, ensure that we really are making a change:
+          assert.notDeepEqual(targetBalance, existingBalance);
```

Comment:
I would avoid making deep equal comparisons of BN objects. It's probably fine for now, but it seems somewhat brittle to me. Instead, I'd use `assert.isFalse(targetBalance.eq(existingBalance))`

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/2013b5abe2c2cf6ac80f144ccdf24db27e755551